src/Microsoft.AspNetCore.ResponseCompression/BrotliCompressionProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.ResponseCompression
+{
+    /// <summary>
+    /// Brotli compression provider.
+    /// </summary>
+    public class BrotliCompressionProvider : ICompressionProvider
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="BrotliCompressionProvider"/> with options.
+        /// </summary>
+        /// <param name="options"></param>
+        public BrotliCompressionProvider(IOptions<BrotliCompressionProviderOptions> options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            Options = options.Value;
+        }
+
+        private BrotliCompressionProviderOptions Options { get; }
+
+        /// <inheritdoc />
+        public string EncodingName => "br";
+
+        /// <inheritdoc />
+        public bool SupportsFlush => true;
+
+        /// <inheritdoc />
+        public Stream CreateStream(Stream outputStream)
+        {
+#if NET461
+            throw new PlatformNotSupportedException();
```

Comment:
Do these require a message?

Reply:
I'm not worried about it. I don't expect people to hit this.
You can combine net461 and netstandard2_0.

Reply:
Not sure if that would work. I think you need to check NETCOREAPP2_1 first, then check NETSTANDARD2_0

Reply:
Nvrm, I didn't realize ifdefs are not inherited.